src/mlpack/core/util/arma_config.hpp

```diff
@@ -15,8 +15,18 @@
 #ifndef MLPACK_CORE_UTIL_ARMA_CONFIG_HPP
 #define MLPACK_CORE_UTIL_ARMA_CONFIG_HPP
 
-#define MLPACK_ARMA_NO64BIT_WORD
+#ifdef ARMA_64BIT_WORD
```

Comment:
This file is auto-generated while building mlpack, and this file is system dependent. But we are building the binding using R and not cmake, Hence we have to take care of this file on our own. I have tried to config this, please guide me whether my approach is correct or not.

Reply:
Right, the only reason this file exists is to warn the user if they compiled `libmlpack.so` with `ARMA_64BIT_WORD` set but are now compiling something *against* `libmlpack.so` *without* that define set (or vice versa).  But since we are compiling *everything* with R, all at once, at the time that `install.packages()` is called (or similar), then this situation isn't possible.  To be honest, I think that you could work around it by leaving the file as-is, and then simply "cheating" and doing

```
#define MLPACK_CORE_ARMA_UTIL_CONFIG_HPP
```

in `RcppMLPACK.h` before including `mlpack/core.hpp` or another mlpack header.  That way, when this file is encountered, it will simply be skipped.  This way, we avoid displaying any spurious warnings, *and* you don't need to do any modification of mlpack's sources. :+1:

Reply:
Yes, sounds legit. Many moons ago we also sorted the (simpler) case of alwasy calling (Rcpp)Armadillo headers before Rcpp headers. We control the space and build---so we can just define as needed.

Reply:
This is working great thank you. I will also update this in the main(Yashwants19/mlpack) repository.